src/features/products/screen/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { FlatList, ScrollView, View } from 'react-native';
+
+import { useGetProducts } from 'network/queries/product-queries';
+
+import ProductItem from './productItem';
+import styles from './styles';
+
+const PurchasesScreen = () => {
+  const { data } = useGetProducts();
+  const items = data?.data.length || 0;
+
+  return (
+    <ScrollView contentContainerStyle={styles.container} showsVerticalScrollIndicator={false}>
+      <View style={styles.flexContainer}>
+        <FlatList
+          scrollEnabled
+          showsHorizontalScrollIndicator={false}
+          data={data?.data}
+          keyExtractor={product => product.id}
+          renderItem={({ item, index }) => {
+            return ProductItem({ item, isLast: index === items - 1 });
```

Comment:
Here you can implicitly return `ProductItem`

``` suggestion
            ProductItem({ item, isLast: index === items - 1 });
```

Reply:
No matter if the data object is undefined or if doesn't hold an array, with the use of `?` you're covered.
So, in the case you don't have any products `isLast` will be false.

```suggestion
            return ProductItem({ item, isLast: index === data?.length - 1 });
```

Reply:
this will only work once I refactor the useGetProducts. Thanks for the suggestion